test/sequential/test-watch-mode.mjs

```diff
@@ -356,4 +358,83 @@ console.log(values.random);
       `Completed running ${inspect(file)}`,
     ]);
   });
+
+  it('should pass IPC messages from a spawning parent to the child and back', async () => {
+    const file = createTmpFile(`console.log('running');
+process.on('message', (message) => {
+  if (message === 'exit') {
+    process.exit(0);
+  } else {
+    console.log('Received:', message);
+    process.send(message);
+  }
+})`);
+
+    const child = spawn(
```

Comment:
can we reuse `failWriteSucceed` or `runWriteSucceed`?

Reply:
can't we just validate that all sent messages are recieved and printed to stdout?

Reply:
If we only care about parent -> child, that would work, but most of the changes in this PR (detroyIPC and associated changes) were to support child -> parent, specifically after restart. Without the echo of the child, this wouldn't be tested (in fact I only figured out it wasn't working because of this test).